hyperspy/_signals/signal1d.py

```diff
@@ -1435,7 +1438,10 @@ def find_peaks1D_ohaver(self, xdim=None,
                          parallel=parallel,
                          max_workers=max_workers,
                          inplace=False)
-        return peaks.data
+        if peaks._lazy:
```

Comment:
It's just that the peaks are often used in other functions and this breaks them later on.  They were being computed before but kind of behind the scenes.  Honestly the peak finding stuff is a pain and keeps breaking.  There are lots of little cases where the behavior of the map function isn't really as advertised.  I can make this change but it will break a couple of tests and maybe some people's codes... If this is just a patch release I thought that we might want to hold off until at least the next minor release to make some changes there

Reply:
> It's just that the peaks are often used in other functions and this breaks them later on. They were being computed before but kind of behind the scenes. Honestly the peak finding stuff is a pain and keeps breaking.

This is confusing: where was it computed before?

Reply:
I changed this so it isn't computed.  The error was just in how the find peaks method is using custom numpy dtypes to save values like the height, width etc.  I don't know if that is really the best way to be saving these values but that might be beside the point and something to look into later. With how ragged signals are saved with a numpy object dtype selecting values beased on dtpye from a dask array no longer works and you have to compute the lazy array before acessing the value.